### PR TITLE
Basic Bleeding Icon + Basic Bleeding Refactor

### LINKED ISSUE #7629: bleeding/low blood level warning icon B:75
### Description

This bounty calls for the implementation of the bleeding status icon. With the more advanced bloodstream model added to the game in the health refactor and the trauma system additions, it's crucial to deliver better feedback to players about how badly they are bleeding.

This warning should evaluate all sources of bleeding occurring in a character's body and escalate the bleed icon as needed to communicate the total bleed rate, and update or clear the alert as the bleeding rate increases or decreases.

Once a character's blood level is particularly low, the blood icon lingers on the screen even if there is no active bleeding, though it no longer blinks bright red if there is no active bleeding.

here is the most up-to-date version of the bleed icons
bleedicon

## What is needed to fulfill the bounty:

- [x] add the bleeding warning to the game.

### PR DESCRIPTION
**Pull request by VernaSaurus**
CL: [Improvement] Bleeding now uses "bleedstacks" similar to firestacks. In addition, a bleeding indicator shows up to show how badly you're bleeding.

I don't have a recording software so I can't show it in action sorry

Closes #7629

### CODE REVIEW

**Comment by PerfectTangent** [comment]
DNM, waiting for some requested tweaks.

**Comment by VernaSaurus** [comment]
waiting on some people to review this please

**Comment by corp-0** [comment]
It has a DNM label, is it ready to review now? Is the discussion over?

**Comment by VernaSaurus** [comment]
I made the requested tweaks, so I assume so.

File: UnityProject/Assets/Scripts/HealthV2/Living/LivingHealthMasterBase.cs
Comment (Bod9001) [inline UnityProject/Assets/Scripts/HealthV2/Living/LivingHealthMasterBase.cs]: Whatever jumps directly from 3 to 5?

Comment (VernaSaurus) [reply UnityProject/Assets/Scripts/HealthV2/Living/LivingHealthMasterBase.cs]: It's just how switch statements work. That basically means "4 or 5"

Comment (ThatDan123) [reply UnityProject/Assets/Scripts/HealthV2/Living/LivingHealthMasterBase.cs]: Looks bad, cant you just use Or || ?

Comment (Bod9001) [reply UnityProject/Assets/Scripts/HealthV2/Living/LivingHealthMasterBase.cs]: sorry *what happens if it jumps from 3 to 5?

Comment (Bod9001) [reply UnityProject/Assets/Scripts/HealthV2/Living/LivingHealthMasterBase.cs]: *2
skipping 3

Comment (VernaSaurus) [reply UnityProject/Assets/Scripts/HealthV2/Living/LivingHealthMasterBase.cs]: Not much, it just does the 5 case

> Looks bad, cant you just use Or || ?

Nope. Trust me, I tried, switch statements literally don't have an "or" case unless you do this.

File: UnityProject/Assets/Scripts/HealthV2/Living/LivingHealthMasterBase.cs
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/HealthV2/Living/LivingHealthMasterBase.cs]:
```suggestion
			switch (Math.Ceiling(bleedStacks))
			{
				case 0:
					State = BleedingState.None;
					break;
				case 1:
					State = BleedingState.VeryLow;
					break;
				case int n when n.IsBetween(2, 3):
					State = BleedingState.Low;
					break;
				case int n when n.IsBetween(4, 6):
					State = BleedingState.Medium;
					break;
				case int n when n.IsBetween(7, 8):
					State = BleedingState.High;
					break;
				case int n when n.IsBetween(9, 10):
					State = BleedingState.UhOh;
					break;
```

Would also work

Comment (VernaSaurus) [reply UnityProject/Assets/Scripts/HealthV2/Living/LivingHealthMasterBase.cs]: that does look better, thanks for the suggestion

**Comment by ThatDan123** [comment]
Also codacy is complaining about PascalCase on `bleedStacks => ` . Should be `BleedStacks => `

**Comment by VernaSaurus** [comment]
> Also codacy is complaining about PascalCase on `bleedStacks => ` . Should be `BleedStacks => `

the problem is that there's already a BleedStacks, which is why I can't change it

File: UnityProject/Assets/Scripts/HealthV2/Living/LivingHealthMasterBase.cs
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/HealthV2/Living/LivingHealthMasterBase.cs]:
```suggestion
		/// <summary>
		/// How badly we're bleeding, same as tg bleed_stacks. 0 = not bleeding.
		/// Exists client side - synced with server.
		/// </summary>
		// FireStacks note: It's called "stacks" but it's really just a floating point value that
		// can go up or down based on possible sources of being on fire. Max seems to be 20 in tg.
		public float BleedStacks => healthStateController.BleedStacks;
```

You'll need to replace all the `bleedStacks` with `BleedStacks` in this script

File: UnityProject/Assets/Scripts/Items/Medical/HealsTheLiving.cs
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Items/Medical/HealsTheLiving.cs]:
```suggestion
				livingHealth.HealthStateController.SetBleedStacks(0f);
```

File: UnityProject/Assets/Scripts/HealthV2/Living/LivingHealthMasterBase.cs
Comment (VernaSaurus) [reply UnityProject/Assets/Scripts/HealthV2/Living/LivingHealthMasterBase.cs]: this caused it to completely break lmao

**Review by Bod9001 — DISMISSED** [review]

File: UnityProject/Assets/Scripts/HealthV2/Living/LivingHealthMasterBase.cs
Comment (corp-0) [inline UnityProject/Assets/Scripts/HealthV2/Living/LivingHealthMasterBase.cs]: can't you simply add with a ``Mathf.Clamp`` to make sure it isn't less than zero and also more than the max?

Comment (VernaSaurus) [reply UnityProject/Assets/Scripts/HealthV2/Living/LivingHealthMasterBase.cs]: Alright, I'll add that to firestacks as well (since they also do this)

Comment (VernaSaurus) [reply UnityProject/Assets/Scripts/HealthV2/Living/LivingHealthMasterBase.cs]: ok, done

**Review by Bod9001 — APPROVED** [review]

**Comment by VernaSaurus** [comment]
just waiting on one more approval

**Review by ThatDan123 — APPROVED** [review]

**Pull request merged**
